### PR TITLE
feat(vector): add set viewport support for vector api with thorvg

### PR DESCRIPTION
**Pull request by onecoolx**
### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by FASTSHIFT — DISMISSED** [review]

File: src/libs/thorvg/tvgCapi.cpp [hunk 1]
```diff
@@ -141,6 +141,12 @@ TVG_API Tvg_Result tvg_canvas_sync(Tvg_Canvas* canvas)
     return (Tvg_Result) reinterpret_cast<Canvas*>(canvas)->sync();
 }
 
+TVG_API Tvg_Result tvg_canvas_set_viewport(Tvg_Canvas* canvas, int32_t x, int32_t y, int32_t w, int32_t h)
```
Comment (XuNeo) [inline src/libs/thorvg/tvgCapi.cpp:144]: Let's upstream this change to ThorVG.

File: src/libs/thorvg/tvgStr.cpp [hunk 2]
```diff
@@ -186,7 +186,7 @@ float strToFloat(const char *nPtr, char **endPtr)
         auto scale = 1.0f;
 
         while (exponentPart >= 8U) {
-            scale *= 1E8;
+            scale *= (float)1E8;
```
Comment (XuNeo) [inline src/libs/thorvg/tvgStr.cpp]: Following the style in this file.
```
        auto scale = 1.0f;
```
Let's add `f` suffix instead.

```suggestion
            scale *= 1E8f;
```

Comment (onecoolx) [reply src/libs/thorvg/tvgStr.cpp]: done

File: src/libs/thorvg/tvgCapi.cpp (hunk 1 above)
Comment (onecoolx) [reply src/libs/thorvg/tvgCapi.cpp:144]: thorvg  already fixed yet

Comment (onecoolx) [reply src/libs/thorvg/tvgCapi.cpp:144]: https://github.com/thorvg/thorvg/commit/178cf2944d5d5176a80f32df3916aab866935dbc

**Review by FASTSHIFT — APPROVED** [review]

**Review by XuNeo — APPROVED** [review]

**Pull request merged**